Sources/CSRF/CSRFFormFieldTag.swift

```diff
@@ -0,0 +1,23 @@
+import Vapor
+import Leaf
+
+public struct CSRFFormFieldTag: LeafTag {
+
+	public init() {}
+
+	public func render(_ ctx: LeafContext) throws -> LeafData {
+
+		struct CSRFFormFieldTagError: Error {}
+
+		try ctx.requireParameterCount(0)
+
+		guard let req = ctx.request else {
+			throw CSRFFormFieldTagError()
```

Comment:
Since this is thrown because `context.request == nil`, then why not have a specific error name? It will help callers debug this scenario.

```
extension CSRFFormFieldTag {
  public enum Error: Swift.Error {
    case nilContextRequest
  }
}
```